commerce-service/src/main/java/com/liferay/commerce/internal/order/CommerceOrderHttpHelperImpl.java

```diff
@@ -422,8 +426,9 @@ private CommerceOrder _getUserCurrentCommerceOrder(
 			themeDisplay.getRequest(), cookieName, true);
 
 		if (Validator.isNotNull(commerceOrderUuid)) {
-			commerceOrder = _commerceOrderService.fetchCommerceOrder(
```

Comment:
@Commando18 Why you are using local service? We must ensure that the order is of the right customer in this method.